Error Message on clicking show details should display detailed error message instead of [object object]

Sharvaribarve opened the pull request:
In case, the validation of resource file fails, the show details button on clicking displays [object object], this has been changed to display detailed error message.

Sharvaribarve committed:
Iterating over content object to display individual properties

jmahmoodcc reviewed (changes requested):
Looks good, just clean up the formatting please (the indenting is off)

Sharvaribarve committed:
Formmated code

Sharvaribarve committed:
Added line break

jeddcuison reviewed (changes requested):
In lieu of Sharvari's findings that the details are the same as the message content displayed by the error dialog, the solution as discussed with the team is just to display the message. My bad I did not see this coming, the details section is supposed to be for stacktrace details. Please revert the changes and just omit the content parameter in $.errorAlert. By not providing the content, errorAlert is smart enough not to display the details button. Thanks!

Sharvaribarve committed:
Reverted changes and added to check if content is an array

jeddcuison commented on jwala-webapp/src/main/webapp/resources/js/react/ResourcePane.js:
```diff
@@ -166,7 +166,7 @@ var ResourcePane = React.createClass({
     },
     displayDeployErrorMessage: function(response){
-        $.errorAlert(ResourcePane.parseDetailedErrorMsg(response, ResourcePane.DEFAULT_DEPLOY_ERR_MSG), "", undefined, response.responseJSON.content);
+        $.errorAlert(ResourcePane.parseDetailedErrorMsg(response, ResourcePane.DEFAULT_DEPLOY_ERR_MSG), "");
```
Please revert to:

    $.errorAlert(ResourcePane.parseDetailedErrorMsg(response, ResourcePane.DEFAULT_DEPLOY_ERR_MSG), "", undefined, response.responseJSON.content);

jeddcuison commented on jwala-webapp/src/main/webapp/resources/js/error-alert.js:
```diff
@@ -45,15 +45,7 @@ $.extend({ errorAlert: function (message, dlgTitle, modal, content) {
-                    }
-                }
-                content = details;
+            if (content && content !== "" && Array.isArray(content)) {
```
Please remove unnecessary content validation, this will do 

    if (Array.isArray(content)) {
        ...
    }

jeddcuison reviewed (changes requested):
Almost there, 2 additional comments, after those are addressed then we're good to go

Sharvaribarve committed:
Reverted changes in Resource  and removed extra content check in error alert

jeddcuison reviewed (commented):
The last change looks good. Can you write a selenium test for this ?

jeddcuison committed:
Add test case for deployment of an invalid resource

Check if the error dialog box that is displayed after trying to deploy an invalid resource has the details section (the details section should not be shown)
